src/vm/mod.rs

```diff
@@ -1,3 +1,115 @@
 // SPDX-License-Identifier: Apache-2.0
 
 mod linux;
+
+use crate::vm::linux::{
+    ioctl::{Cmd, CmdId},
+    types::{Capabilities, CpuidConfig, NR_CPUID_CONFIGS},
+};
+
+/// Handle to the KVM VM file descriptor
+pub struct Vm(pub kvm_ioctls::VmFd);
+
+impl Vm {
+    /// Retrieve information about the Intel TDX module
+    ///
+    /// # Example:
+    /// ```
+    /// use kvm_ioctls::Kvm;
+    /// use tdx::vm::Vm;
+    ///
+    /// let kvm_fd = Kvm::new().unwrap();
+    /// let vm_fd = kvm_fd.create_vm_with_type(2).unwrap();
+    /// let caps = Vm(vm_fd).get_capabilities().unwrap();
+    ///
+    /// println!("{:#?}", caps);
+    /// ```
+    pub fn get_capabilities(&self) -> Result<TDXCapabilities, kvm_ioctls::Error> {
```

Comment:
For now, this API could take one `u32` input indicating the number of CPUID configs. We could remove this if a stable number of configs becomes available.